### ISSUE TITLE
Results of multi-step denoise and single-step denoise

### ISSUE THREAD

**Issue by OrangeSodahub** [JunzheJosephZhu/HiFA#12]
Hi, I notice that in your code you have both options of multi-steps denoise and single-step denoise. I wonder that the results showed in your paper belongs to which one? If it is single-step denoise, so how was the results of multi-steps denoise? Did multi-step denoise better than single-step denoise?

 Sorry after many tries I still cannot run your code in my machine so didn't validate myself. Thanks!

**Comment by HiFA-team** [comment]
I think for some prompts it was 4 step, for some it was single-step. Doing
multistep denoising between 1-6 steps didn't cause a huge difference.
Increasing the step number to above 10 caused a huge difference, the
generated object were more color shifted and shrinked, and even more so if
I use a DPM solver.

This paper is also implemented on threestudio.

Regarding not being able to run the code, maybe check this out:
https://github.com/JunzheJosephZhu/HiFA/issues/9#issuecomment-2056222628

**Comment by OrangeSodahub** [comment]
Were you able to get right rgb image from single denoise step using ddim? I always get noisy image following `pred_original_sample = (sample - beta_prod_t ** (0.5) * model_output) / alpha_prod_t ** (0.5)`

**Comment by OrangeSodahub** [comment]
@HiFA-team Hi, thanks for your information. Do you have any examples showing 'color-shifting' or 'shrink', thanks!

**Comment by HiFA-team** [comment]
Nope, single ddim denoising step just doesnt work. There are two problems
here: 1. DDIM typically only works with >10 steps. 2. CFG=100 will make
this worse, although the pseudo GT is much more clear when num_steps>20. I
feel like having many more steps just cancels the effect caused by
CFG=100(I like to think of DDIM as an SDE solver, and CFG to be some sort
of error/noise that happened in the solving process when the integrator's
step size is too big)
